addons/web_editor/static/src/js/editor/odoo-editor/src/utils/sanitize.js

```diff
@@ -79,186 +73,125 @@ export function areSimilarElements(node, node2) {
-                if (restoreCursor) {
-                    restoreCursor();
-                }
-            }
-
-            // Remove empty blocks in <li>
-            if (
-                node.nodeName === 'P' &&
-                node.parentElement.tagName === 'LI' &&
-                isEmptyBlock(node)
-            ) {
-                const parent = node.parentElement;
-                const restoreCursor = node.isConnected &&
-                    preserveCursor(this.root.ownerDocument);
-                node.remove();
-                fillEmpty(parent);
-                if (restoreCursor) {
-                    restoreCursor(new Map([[node, parent]]));
-                }
-            }
-
-            // Transform <li> into <p> if they are not in a <ul> / <ol>
-            if (node.nodeName === 'LI' && !node.closest('ul, ol')) {
-                const paragraph = document.createElement("p");
-                paragraph.replaceChildren(...node.childNodes);
-                node.replaceWith(paragraph);
-                node = paragraph;
-            }
-
-            // Ensure a zero width space is present inside the FA element.
-            if (isFontAwesome(node) && node.textContent !== '\u200B') {
-                node.textContent = '\u200B';
             }
+        }
+    }
+    return node;
+}
 
-            // Ensure the editor tabs align on a 40px grid.
-            if (isEditorTab(node)) {
-                let tabPreviousSibling = node.previousSibling;
-                while (isZWS(tabPreviousSibling)) {
-                    tabPreviousSibling = tabPreviousSibling.previousSibling;
-                }
-                if (isEditorTab(tabPreviousSibling)) {
-                    node.style.width = '40px';
-                } else {
-                    const editable = closestElement(node, '.odoo-editor-editable');
-                    if (editable && editable.firstElementChild) {
-                        const nodeRect = node.getBoundingClientRect();
-                        const referenceRect = editable.firstElementChild.getBoundingClientRect();
-                        // Values from getBoundingClientRect() are all zeros
-                        // during Editor startup or saving. We cannot
-                        // recalculate the tabs width in thoses cases.
-                        if (nodeRect.width && referenceRect.width) {
-                            const width = (nodeRect.left - referenceRect.left) % 40;
-                            node.style.width = (40 - width) + 'px';
-                        }
-                    }
-                }
-            }
+export function sanitize(root) {
+    const start = root.ownerDocument.getSelection()?.anchorNode;
+    const rootClosestBlock = closestBlock(root);
+    if (rootClosestBlock) {
+        // Sanitize the tree.
+        const isList = ['UL', 'OL'].includes(rootClosestBlock.nodeName);
+        let node = isList ? rootClosestBlock.parentElement : rootClosestBlock;
```

Comment:
This is worth its own comment as well IMO as it is not obvious to the reader why lists are a special case.

Reply:
```js
// If the node is a list, start sanitization from its parent to ensure
// adjacent lists are merged when needed.
```
?